Mathlib/FieldTheory/KummerExtension.lean

```diff
@@ -0,0 +1,444 @@
+/-
+Copyright (c) 2023 Andrew Yang, Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.RootsOfUnity.Basic
+import Mathlib.RingTheory.AdjoinRoot
+import Mathlib.LinearAlgebra.Charpoly.Basic
+import Mathlib.FieldTheory.Galois
+import Mathlib.LinearAlgebra.Eigenspace.Minpoly
+/-!
+# Kummer Extensions
+
+## Main result
+- `isCyclic_tfae`:
+Suppose `L/K` is a finite extension of dimension `n`, and `K` contains all `n`-th roots of unity.
+Then `L/K` is cyclic iff
+`L` is a splitting field of some irreducible polynomial of the form `Xⁿ - a : K[X]` iff
+`L = K[α]` for some `αⁿ ∈ K`.
+
+- `autEquivRootsOfUnity`:
+Given an instance `IsSplittingField K L (X ^ n - C a)`
+(perhaps via `isSplittingField_X_pow_sub_C_of_root_adjoin_eq_top`),
+then the galois group is isomorphic to `rootsOfUnity n K`, by sending
+`σ ↦ σ α / α` for `α ^ n = a`, and the inverse is given by `μ ↦ (α ↦ μ • α)`.
+
+- `autEquivZmod`:
+Furthermore, given an explicit choice `ζ` of a primitive `n`-th root of unity, the galois group is
+then isomorphic to `Multiplicative (ZMod n)` whose inverse is given by
+`i ↦ (α ↦ ζⁱ • α)`.
+
+## Other results
+Criteria for `X ^ n - C a` to be irreducible is given:
+`X_pow_sub_C_irreducible_iff_of_prime`: `X ^ n - C a` is irreducible iff `a` is not a `p`-power.
+
+TODO: criteria for general `n`.
+
+-/
+variable {K : Type*} [Field K]
+
+open Polynomial IntermediateField AdjoinRoot
+
+section Splits
+
+lemma root_X_pow_sub_C_pow (n : ℕ) (a : K) :
+    (AdjoinRoot.root (X ^ n - C a)) ^ n = AdjoinRoot.of _ a := by
+  rw [← sub_eq_zero, ← AdjoinRoot.eval₂_root, eval₂_sub, eval₂_C, eval₂_pow, eval₂_X]
+
+lemma root_X_pow_sub_C_ne_zero {n : ℕ} (hn : 1 < n) (a : K) :
+    (AdjoinRoot.root (X ^ n - C a)) ≠ 0 :=
+  mk_ne_zero_of_natDegree_lt (monic_X_pow_sub_C _ (Nat.not_eq_zero_of_lt hn))
+    X_ne_zero <| by rwa [natDegree_X_pow_sub_C, natDegree_X]
+
+lemma root_X_pow_sub_C_ne_zero' {n : ℕ} {a : K} (hn : 0 < n) (ha : a ≠ 0) :
+    (AdjoinRoot.root (X ^ n - C a)) ≠ 0 := by
+  obtain (rfl|hn) := (Nat.succ_le_iff.mpr hn).eq_or_lt
+  · rw [← Nat.one_eq_succ_zero, pow_one]
+    intro e
+    refine mk_ne_zero_of_natDegree_lt (monic_X_sub_C a) (C_ne_zero.mpr ha) (by simp) ?_
+    trans AdjoinRoot.mk (X - C a) (X - (X - C a))
+    · rw [sub_sub_cancel]
+    · rw [map_sub, mk_self, sub_zero, mk_X, e]
+  · exact root_X_pow_sub_C_ne_zero hn a
+
+lemma injOn_pow_mul_of_isPrimitiveRoot {n : ℕ} (hn : 0 < n) {ζ : K} (hζ : IsPrimitiveRoot ζ n)
```

Comment:
This can go in the file about roots of unity (and again it doesn't need `Field K`).